hints/eating-bananas.md

```diff
@@ -0,0 +1,47 @@
+<br>
+<details class="hint-accordion">  
+    <summary>Recommended Time & Space Complexity</summary>
+    <p>
+    You should aim for a solution with <code>O(nlogm)</code> time and <code>O(1)</code> space, where <code>n</code> is the size of the input array, and <code>m</code> is the maximum value in the array.
+    </p>
+</details>
+
+<br>
+<details class="hint-accordion">  
+    <summary>Hint 1</summary>
+    <p>
+    Can you determine the upper bound for the possible answer? How much time does it take Koko to eat a pile with <code>x</code> bananas? 
+    </p>
+</details>
+
+<br>
+<details class="hint-accordion">  
+    <summary>Hint 2</summary>
+    <p>
+    It takes <code>ceil(x / k)</code> time to finish the <code>x</code> pile when Koko eats at a rate of <code>k</code> bananas per hour. Our task is to determine the minimum possible value of <code>k</code>. However, we must also ensure that at this rate, <code>k</code>, Koko can finish eating all the piles within the given <code>h</code> hours. Can you now think of the upper bound for <code>k</code>?
+    </p>
+</details>
+
+<br>
+<details class="hint-accordion">  
+    <summary>Hint 3</summary>
+    <p>
+    The upper bound for <code>k</code> is the maximum size of all the piles. Why? Because if Koko eats largest pile in one hour, then it is straightforward that she can eat any other pile in an hour only.
```

Comment:
> Because if Koko eats largest pile in one hour, then it is straightforward that she can eat any other pile in an hour only.

if koko eats [the] largest pile

Now that we have tools like chatgpt to proof read, I don't really wanna catch grammar errors for you.

Reply:
It is weird that after I wrote the hint and cross-checked it with GPT for errors, it can also make mistakes in grammar.